File: Ruby/problem6.rb
def sum_square_difference(n)
  sum_of_squares = (n ** 3.0 / 3.0) + (n ** 2.0 / 2.0) + (n / 6.0)
  square_of_sum = ((n * (n + 1.0)) / 2.0) ** 2.0
  difference = square_of_sum - sum_of_squares
  difference === Float::INFINITY ? difference : difference.to_i
end

max = ARGV.first ? ARGV.first.to_i : 10
start_time = Time.now
puts sum_square_difference(max)
puts "#{Time.now - start_time} seconds."

File: C#/problem6.cs
using System;
using System.Diagnostics;

class Problem6
{
  public static void Main(string[] args)
  {
    int max = 10;
    if (args.Length > 0)
    {
      max = Convert.ToInt32(args[0]);
    }

    Stopwatch stopwatch = new Stopwatch();
    stopwatch.Start();
    Console.WriteLine(SumSquareDifference(max));
    stopwatch.Stop();
    Console.WriteLine(stopwatch.ElapsedMilliseconds + " seconds");
  }

  public static int SumSquareDifference(int n)
  {
    double sumOfSquares = (n * n * n / 3.0) + (n * n / 2.0) + (n / 6.0);
    double squareOfSum = ((n * (n + 1.0)) / 2.0 ) * ((n * (n + 1.0)) / 2.0 );
    double difference = squareOfSum - sumOfSquares;
    return Convert.ToInt32(difference);
  }
}
